openedx/core/djangoapps/catalog/management/commands/cache_course_runs.py

```diff
@@ -0,0 +1,147 @@
+
+            logger.info(f'Caching details for {courserun_keys} courses.')
+            # TODO: Cache course details by course run key
+            cache.set_many(processed_courses_details, None)
+
+        if failure:
+            sys.exit(1)
+
+    @classmethod
+    def chunks(cls, keys, chunk_size=50):
+        """
+        Yield chunks of size `chunk_size`
+        """
+        for i in range(0, len(keys), chunk_size):
+            yield keys[i:i + chunk_size]
+
+    @classmethod
+    def fetch_courses_details(cls, client, courserun_keys, api_base_url):  # lint-amnesty, pylint: disable=missing-function-docstring
+        """
+        Fetch the course data from discovery using /api/v1/courses endpoint
+        """
+        course_keys = [cls.construct_course_key(courserun_key) for courserun_key in courserun_keys]
+        encoded_course_keys = ','.join(map(quote_plus, course_keys))
+
+        logger.info(f'Fetching details for courses {course_keys}.')
+        api_url = f"{api_base_url}/courses"
+        response = client.get(api_url, params={"keys": encoded_course_keys})
+        response.raise_for_status()
+        courses_details = response.json()
+
+        return courses_details
+
+    @classmethod
+    def process_courses_details(cls, courserun_keys, courses_details):
+        """
+        Parse and extract the minimal data that we need.
+        """
+        courses = {}
+        for courserun_key in courserun_keys:
+            course_metadata = cls.find_attr(courses_details, 'key', courserun_key)
+
+            course_type = course_metadata.get('course_type')
+            product_source = course_metadata.get('product_source')
+            if product_source:
+                product_source = product_source.get('slug')
+
+            if course_type == EXEC_ED_COURSE_TYPE:
+                additional_metadata = course_metadata.get('additional_metadata')
+                if additional_metadata:
+                    enroll_by = additional_metadata.get('registration_deadline')
+                    start_date = additional_metadata.get('start_date')
+                    end_date = additional_metadata.get('end_date')
+            else:
+                course_run = cls.find_attr(course_metadata.get('course_runs'), 'key', courserun_key)
+                seats = course_run.get('seats')
+                # TODO: sort the seats according to best mode order and pick first
+                # seat = sorted(course_run.get('seats'), key=lambda s: seats.index(s), reverse=True)
+                seat = course_run.get('seats')[0]
+                enroll_by = seat.get('upgrade_deadline')
+                start_date = course_run.get('start')
+                end_date = course_run.get('end')
+
+            course = {
+                'course_type': course_type,
+                'product_source': product_source,
+                'enroll_by': enroll_by,
+                'start_date': start_date,
+                'end_date': end_date,
```

Comment:
aee comment above ... would appreciate if you can add the COURSE_HOST_URL = "https://www.getsmarter.com/" here ...